source/portable/mbedtls/core_pkcs11_mbedtls.c

```diff
@@ -3028,10 +3006,26 @@ CK_DECLARE_FUNCTION( CK_RV, C_FindObjectsInit )( CK_SESSION_HANDLE hSession,
         {
             xAttribute = pTemplate[ ulIndex ];
 
-            if( xAttribute.type == CKA_LABEL )
+            if( ( xAttribute.type == CKA_LABEL ) && ( xAttribute.ulValueLen <= pkcs11configMAX_LABEL_LENGTH ) )
             {
-                ( void ) memcpy( pxSession->pxFindObjectLabel, xAttribute.pValue, xAttribute.ulValueLen );
-                xResult = CKR_OK;
+                /* Plus one to leave room for a NULL terminator. */
+                pxFindObjectLabel = mbedtls_calloc( 1, pTemplate->ulValueLen + 1UL );
+
+                if( pxFindObjectLabel != NULL )
+                {
+                    pxSession->xFindObjectLabelLen = pTemplate->ulValueLen;
+                    pxSession->pxFindObjectLabel = pxFindObjectLabel;
+                    ( void ) memset( pxFindObjectLabel, 0, pTemplate->ulValueLen + 1UL );
```

Comment:
Doesn't `calloc` already set the allocated memory to zero? Don't think this `memset` is necessary